File: dotnet/src/CarespaceSDK/CarespaceClient.cs
using Microsoft.Extensions.Logging;
using CarespaceSDK.Api;
using CarespaceSDK.Configuration;
using CarespaceSDK.Http;
using CarespaceSDK.Models;

namespace CarespaceSDK;

public class CarespaceClient : IDisposable
{
    private readonly IHttpClient _httpClient;
    private readonly CarespaceConfiguration _configuration;
    private bool _disposed;

    public IAuthApi Auth { get; }
    public IUsersApi Users { get; }
    public IClientsApi Clients { get; }
    public IProgramsApi Programs { get; }

    public CarespaceClient(CarespaceConfiguration? configuration = null, ILogger<CarespaceHttpClient>? logger = null)
    {
        _configuration = configuration ?? CarespaceConfiguration.ForDevelopment();
        _configuration.Validate();

        var httpClient = new HttpClient();
        _httpClient = new CarespaceHttpClient(httpClient, _configuration, logger);

        Auth = new AuthApi(_httpClient);
        Users = new UsersApi(_httpClient);
        Clients = new ClientsApi(_httpClient);
        Programs = new ProgramsApi(_httpClient);
    }

    public CarespaceClient(string apiKey, string? baseUrl = null, ILogger<CarespaceHttpClient>? logger = null)
        : this(new CarespaceConfiguration 
        { 
            ApiKey = apiKey, 
            BaseUrl = baseUrl ?? "https://api-dev.carespace.ai" 
        }, logger)
    {
    }

    public void SetApiKey(string apiKey)
    {
        _httpClient.SetApiKey(apiKey);
    }

    public void SetDefaultHeader(string name, string value)
    {
        _httpClient.SetDefaultHeader(name, value);
    }

    public void RemoveDefaultHeader(string name)
    {
        _httpClient.RemoveDefaultHeader(name);
    }

    public async Task<LoginResponse> LoginAndSetTokenAsync(string email, string password, CancellationToken cancellationToken = default)
    {
        var response = await Auth.LoginAsync(email, password, cancellationToken);
        
        if (response.Success && response.Data != null)
        {
            SetApiKey(response.Data.AccessToken);
            return response.Data;
        }

        throw new InvalidOperationException($"Login failed: {response.Error ?? response.Message}");
    }

    public async Task<PaginatedResponse<User>> QuickGetUsersAsync(int limit = 20, string? search = null, CancellationToken cancellationToken = default)
    {
        return await Users.GetUsersAsync(limit: limit, search: search, cancellationToken: cancellationToken);
    }

    public async Task<PaginatedResponse<Client>> QuickGetClientsAsync(int limit = 20, string? search = null, CancellationToken cancellationToken = default)
    {
        return await Clients.GetClientsAsync(limit: limit, search: search, cancellationToken: cancellationToken);
    }

    public async Task<PaginatedResponse<Program>> QuickGetProgramsAsync(int limit = 20, ProgramCategory? category = null, CancellationToken cancellationToken = default)
    {
        return await Programs.GetProgramsAsync(limit: limit, category: category, cancellationToken: cancellationToken);
    }

    public async Task<bool> HealthCheckAsync(CancellationToken cancellationToken = default)
    {
        try
        {
            if (!string.IsNullOrEmpty(_configuration.ApiKey))
            {
                await Users.GetUserProfileAsync(cancellationToken);
                return true;
            }
            else
            {
                await Users.GetUsersAsync(limit: 1, cancellationToken: cancellationToken);
                return true;
            }
        }
        catch
        {
            return false;
        }
    }

    public static CarespaceClient CreateForDevelopment(string? apiKey = null, ILogger<CarespaceHttpClient>? logger = null)
    {
        return new CarespaceClient(CarespaceConfiguration.ForDevelopment(apiKey), logger);
    }

    public static CarespaceClient CreateForProduction(string apiKey, ILogger<CarespaceHttpClient>? logger = null)
    {
        return new CarespaceClient(CarespaceConfiguration.ForProduction(apiKey), logger);
    }

    public static CarespaceClient CreateForStaging(string apiKey, ILogger<CarespaceHttpClient>? logger = null)
    {
        return new CarespaceClient(CarespaceConfiguration.ForStaging(apiKey), logger);
    }

    public void Dispose()
    {
        if (!_disposed)
        {
            _httpClient?.Dispose();
            _disposed = true;
        }
    }
}

File: temp-repos/carespace-kotlin-sdk/carespace-sdk/src/commonMain/kotlin/com/carespace/sdk/CarespaceClient.kt
package com.carespace.sdk

import com.carespace.sdk.api.*
import com.carespace.sdk.configuration.CarespaceConfiguration
import com.carespace.sdk.http.CarespaceHttpClient
import com.carespace.sdk.models.*

/**
 * Main Carespace SDK client
 */
class CarespaceClient(
    private val configuration: CarespaceConfiguration = CarespaceConfiguration.forDevelopment(),
    httpClient: CarespaceHttpClient? = null
) {
    private val _httpClient: CarespaceHttpClient = httpClient ?: CarespaceHttpClient(configuration)

    // API endpoints
    val auth: AuthApi = AuthApiImpl(_httpClient)
    val users: UsersApi = UsersApiImpl(_httpClient)
    val clients: ClientsApi = ClientsApiImpl(_httpClient)
    val programs: ProgramsApi = ProgramsApiImpl(_httpClient)

    /**
     * Set API key for authentication
     */
    fun setApiKey(apiKey: String) {
        _httpClient.setApiKey(apiKey)
    }

    /**
     * Login and automatically set the access token
     */
    suspend fun loginAndSetToken(email: String, password: String): LoginResponse {
        val response = auth.login(email, password)
        if (response.success && response.data != null) {
            setApiKey(response.data.accessToken)
            return response.data
        } else {
            throw IllegalStateException("Login failed: ${response.error ?: response.message}")
        }
    }

    /**
     * Quick method to get users with minimal parameters
     */
    suspend fun quickGetUsers(limit: Int = 20, search: String? = null): PaginatedResponse<User> {
        return users.getUsers(limit = limit, search = search)
    }

    /**
     * Quick method to get clients with minimal parameters
     */
    suspend fun quickGetClients(limit: Int = 20, search: String? = null): PaginatedResponse<Client> {
        return clients.getClients(limit = limit, search = search)
    }

    /**
     * Quick method to get programs with minimal parameters
     */
    suspend fun quickGetPrograms(limit: Int = 20, category: ProgramCategory? = null): PaginatedResponse<Program> {
        return programs.getPrograms(limit = limit, category = category)
    }

    /**
     * Perform a health check to verify API connectivity
     */
    suspend fun healthCheck(): Boolean {
        return try {
            if (configuration.apiKey != null) {
                users.getUserProfile()
                true
            } else {
                users.getUsers(limit = 1)
                true
            }
        } catch (e: Exception) {
            false
        }
    }

    /**
     * Get current configuration
     */
    fun getConfiguration(): CarespaceConfiguration = configuration

    /**
     * Close the client and cleanup resources
     */
    fun close() {
        _httpClient.close()
    }

    companion object {
        /**
         * Create client for development environment
         */
        fun forDevelopment(apiKey: String? = null): CarespaceClient {
            return CarespaceClient(CarespaceConfiguration.forDevelopment(apiKey))
        }

        /**
         * Create client for production environment
         */
        fun forProduction(apiKey: String): CarespaceClient {
            return CarespaceClient(CarespaceConfiguration.forProduction(apiKey))
        }

        /**
         * Create client for staging environment
         */
        fun forStaging(apiKey: String): CarespaceClient {
            return CarespaceClient(CarespaceConfiguration.forStaging(apiKey))
        }

        /**
         * Create client with custom configuration
         */
        fun withConfiguration(configuration: CarespaceConfiguration): CarespaceClient {
            configuration.validate()
            return CarespaceClient(configuration)
        }

        /**
         * Create client with API key and custom base URL
         */
        fun withApiKey(apiKey: String, baseUrl: String? = null): CarespaceClient {
            val config = CarespaceConfiguration.forDevelopment(apiKey).copy(
                baseUrl = baseUrl ?: "https://api-dev.carespace.ai"
            )
            return CarespaceClient(config)
        }
    }
}